datadog_lambda/wrapper.py

```diff
@@ -190,6 +192,9 @@ def _after(self, event, context):
             status_code = extract_http_status_code_tag(self.trigger_tags, self.response)
             if status_code:
                 self.trigger_tags["http.status_code"] = status_code
+                if not self.already_submitted_errors_metric_before:
```

Comment:
BTW, in your current implementation `already_submitted_errors_metric_before` never resets to `False` once set `True`. `__init__` only gets called once on cold start. You need to reset it in `_before` (or in `_call` before the `try` block)

Reply:
Nice catch. `self.response` is already initialized to `None` here: https://github.com/DataDog/datadog-lambda-python/blob/de5aba059c2c87d83e10622144481bb80f8fe14d/datadog_lambda/wrapper.py#L112-L112 .

And when the exception happens, `self.response` will still be `None`. So, we actually do not need the flag (`self. already_submitted_errors_metric_before `) to avoid sending errors metrics twice since the `statusCode` will be `None`.

Reply:
Yes, it's initialized to None, but it doesn't get reset to None between invocations. E.g., you have a good invocation, which would set self.response, then you have a failed invocation, self.response won't reset to None. extract_http_status_code_tag would extract the status code from the last good invocation instead of from the current failed invocation.

Even putting your use case aside, the state of self.response probably should be reset to None on before every invocation, keeping the invocation response value from the previous invocation in the current invocation is meaningless and can only lead to bugs.

Reply:
Maybe one context you missed here is that Lambda isn't truly stateless, a sandbox/container is reused for many invocations until it gets recycled by AWS. https://pfisterer.dev/posts/aws-lambda-container-reuse might be a good read?

Reply:
Thanks for sharing the post! Just want to be sure (since the post mentioned only about shared folder -- `It's possible that a container is reused, which causes the /tmp folder to be shared between invocations.`), does the singleton wrapper got reused between multiple invocations?
```datadog_lambda_wrapper = _LambdaDecorator```

I plan to reset `self.response` at the beginning of `_before()` anyway. Thanks for calling this out!

Reply:
yes, a new container starts up on first invocation, your python app get loaded into memory, that's called cold start, then from there, the container will be reused for subsequent invocations, until AWS recycles it, or AWS needs to launch more containers to handle more requests (one lambda container can only handle one request/invocation at a time)